src/main/java/com/commercetools/sync/products/templates/beforeupdatecallback/SyncSingleLocale.java

```diff
@@ -141,32 +142,31 @@
                 final String newLocaleValue = newLocalizedField.get(locale);
                 final String oldLocaleValue = oldLocalizedField.get(locale);
                 if (!Objects.equals(newLocaleValue, oldLocaleValue)) {
-                    LocalizedString withLocaleChange = oldLocalizedField;
-                    // if old locale value is set, remove it from old localized field.
-                    if (oldLocaleValue != null) {
-                        withLocaleChange = LocalizedString.of(
+                    // if old locale value is set, remove it from old localized field, otherwise use oldLocalizedField
```

Comment:
I woudn't add the `otherwise use oldLocalizedField`, since there is another check that happens before the `oldLocalizedField` is actually used in the update action:

at the comment after:
`// If old locale value is not set, only update if the new locale value is set.`

Maybe, this bottom comment could be change to, to be more verbose:
`// Only if old locale value is not set and the new locale value is set, update the old localized field with the new locale value .`

Reply:
752be475